docs/guides/streaming/sources/google-cloud-pubsub.mdx

```diff
@@ -0,0 +1,146 @@
+---
+title: "Google Cloud PubSub"
+description:
+  "Ingest data from Google Cloud PubSub event streaming sources."
+---
+
+<img
+  alt="Google Cloud PubSub Source"
+  src="https://lh3.googleusercontent.com/1-WPjO3QIWh0PKPXhHNcKUNcTCeKX218WsC6nIv1xGjGg3VQci-nG9uTqUyBlStJ8YR5LVaC8QFXN6Y7Y9lz=w160-h160"
+/>
+
+<br />
+
+## Config
+
+```yaml
+connector_type: google_cloud_pubsub
+project_id: test-project-id
+topic_id: test-topic-id
+subscription_id: test-subscription-id-0
+timeout: 5
+batch_size: 100
+pubsub_emulator_host: null  # "host.docker.internal:8085"
+path_to_credentials_json_file: "./google_credentials.json"
+```
+
+The `pubsub_emulator_host` parameter is optional, which could be used when you need to test
+the Google Cloud PubSub service locally through an emulator (see more details at the bottom).
+
+## Start Mage
+
+### Create a new pipeline
+
+1. Open Mage in your browser.
+2. Click <b>`+ New pipeline`</b>, then select `Streaming`.
+3. Add a data loader block, select `Google Cloud PubSub`, and paste the following:
+
+    ```yaml
+    connector_type: google_cloud_pubsub
+    project_id: test-project-id
+    topic_id: test-topic-id
+    subscription_id: test-subscription-id-0
+    timeout: 5
+    batch_size: 100
+    pubsub_emulator_host: "host.docker.internal:8085"
```

Comment:
Let's not use the emulator in this example. We should use path_to_credentials_json_file in the example since that's the production use case.

Reply:
Made the suggested change.